lizard_connector/connector.py

```diff
@@ -0,0 +1,342 @@
+    return parse_element(json_, uuid)
+
+
+class Endpoint(object):
+    max_results = 1000
+    endpoint = ""
+
+    def __init__(self, base, **kwargs):
+        """
+        :param base: lizard-nxt url.
+        """
+        self.fetcher = Fetcher(
+            endpoint=self.endpoint,
+            max_results=self.max_results,
+            **kwargs
+        )
+        base = base.strip(r'/')
+        if not base.startswith('http'):
+            base = 'https://' + base
+        base = urllib.parse.urljoin(base, 'api/v2')
+        self.base_url = urllib.parse.urljoin(base, self.endpoint) + "/"
+
+    def get(self, *querydicts, **queries):
+        """
+        Query the api at this endpoint.
+        For possible queries see: https://nxt.staging.lizard.net/doc/api.html
+        Stores the api-response as a dict in the results attribute.
+
+        :param querydicts: all key valuepairs from dictionaries are used as
+            queries.
+        :param queries: all keyword arguments are used as queries.
+        :return:
+        """
+        q = QueryDictionary(page_size=self.max_results)
+        q.update(*querydicts, **queries)
+        query = "?" + urllib.parse.urlencode(q)
+        url = urllib.parse.urljoin(self.base_url, query)
+        return self.fetcher.get(url)
+
+    def post(self, uuid, data):
+        """
+        POST data to the api at this endpoint.
+        :param uuid: UUID of the object in the database you wish to store
+                     data to.
+        :param data: Dictionary with the data to post to the api
+        """
+        self.fetcher.post(self.base_url, uuid, data)
+
+    @property
+    def paginated(self):
+        """
+        Indicates whether this object is paginated (i.e. other pages exist).
+        """
+        return bool(self.fetcher.next_url)
+
+    @property
+    def count(self):
+        """
+        Number of results.
+        """
+        return self.fetcher.count
+
+
+class Annotations(Endpoint):
+    endpoint = 'annotations'
+
+
+class CollageItems(Endpoint):
+    endpoint = 'collageitems'
+
+
+class Collages(Endpoint):
+    endpoint = 'collages'
+
+
+class Dashboards(Endpoint):
+    endpoint = 'dashboards'
+
+
+class Domains(Endpoint):
+    endpoint = 'domains'
+
+
+class Events(Endpoint):
+    endpoint = 'events'
+
+
+class EventSeries(Endpoint):
+    endpoint = 'eventseries'
+
+
+class Favourites(Endpoint):
+    endpoint = 'favourites'
+
+
+class FixedDrainageLevelAreas(Endpoint):
+    endpoint = 'fixeddrainagelevelareas'
+
+
+class Geocode(Endpoint):
+    endpoint = 'geocode'
+
+
+class Inbox(Endpoint):
+    endpoint = 'inbox'
+
+
+class Layers(Endpoint):
+    endpoint = 'layers'
+
+
+class LeveeReferencePoints(Endpoint):
+    endpoint = 'leveereferencepoints'
+
+
+class LeveeRings(Endpoint):
+    endpoint = 'leveerings'
+
+
+class Levees(Endpoint):
+    endpoint = 'levees'
+
+
+class LeveeSections(Endpoint):
+    endpoint = 'leveesections'
+
+
+class LeveeZones(Endpoint):
```

Comment:
Dit zijn een heleboel hardcoded classes. Kan dat niet lekkerder dynamisch? Het hoofd-api-beginpunt aanroepen en dan kijken of iets bestaat? De classes zelf voegen niets toe qua aparte methodes ofzo.